cellmlmanip/units.py

```diff
@@ -162,146 +143,145 @@ def __init__(self, expression):
-        self.ureg.define(definition_string_or_instance)
-        self.custom_defined.add(units_name)
-
     def get_quantity(self, unit_name):
         """Returns a pint `Unit` with the given name from the UnitRegistry.
 
         :param unit_name: string name of the unit
         :returns: `Unit`
         :raises: ``pint.UndefinedUnitError`` if the unit is not present in registry.
         """
+        # TODO CHANGE NAME TO GET UNIT, IF IT RETURNS A UNIT, ELSE UPDATE DOCSTRING
+
         try:
-            return self.ureg.parse_expression(unit_name).units
+            return self._registry.parse_expression(unit_name).units
         except pint.UndefinedUnitError:
             raise KeyError('Cannot find unit <%s> in unit registry' % unit_name)
 
-    def _make_pint_unit_definition(self, units_name, unit_attributes):
-        """
-        Uses a CellML definition to construct a Pint unit definition string.
-
-        :param units_name: The unit name
-        :param unit_attributes: A list of dictionaries. See :meth:`add_custom_unit`.
-        """
-        full_unit_expr = []
-
-        # For each of the <unit> elements for this unit definition
-        for unit_element in unit_attributes:
-            # Source the <unit units="XXX"> from our store
-            matched_unit = self.get_quantity(unit_element['units'])
-
-            # Construct a string representing the expression for this <unit>
-            expr = str(matched_unit)
-
-            # See https://www.cellml.org/specifications/cellml_1.1/#sec_units 5.2.2
-            # offset, prefix, exponent, and multiplier
-
-            if 'prefix' in unit_element:
-                if unit_element['prefix'] in CELLML_UNIT_PREFIXES:
-                    power = CELLML_UNIT_PREFIXES[unit_element['prefix']]
-                else:
-                    # Assume that prefix is an integer - will CellML validation check?
-                    power = '1e%s' % unit_element['prefix']
-                expr = '(%s * %s)' % (expr, power)
-
-            if 'exponent' in unit_element:
-                expr = '((%s)**%s)' % (expr, unit_element['exponent'])
-
-            if 'multiplier' in unit_element:
-                expr = '(%s * %s)' % (unit_element['multiplier'], expr)
-
-            # Collect/add this particular <unit> definition
-            full_unit_expr.append(expr)
-
-        # Join together all the parts of the unit expression
-        full_unit_expr = '*'.join(full_unit_expr)
-
-        # Return Pint definition string
-        logger.debug('Unit %s => %s', units_name, full_unit_expr)
-        return '%s=%s' % (units_name, full_unit_expr)
-
     def is_unit_equal(self, unit1, unit2):
-        """Compares whether two units are equivalent by converting them into base units and comparing the
-        resulting units and multiplicative factors
+        """Compares whether two units are equivalent by converting them into base units and comparing the resulting
```

Comment:
Once we've figured out the best way to compare units, we should make the implementation here match, and use it throughout the code (including CG) avoiding various 'is the conversion factor 1' type tests.